### PR TITLE
Add automated performance test

### LINKED ISSUE #61106: [TECH DEBT] Add performance test to the test suite.
Add performance test to the test suite.

a. start a salt-master, minion 
b. run a state file with multiple states and check the time it takes
c. compare the time with a previous version of salt

### PR DESCRIPTION
**Pull request by MKLeb**
### What does this PR do?
See title

### What issues does this PR fix or reference?
Fixes: https://github.com/saltstack/salt/issues/61106

### Merge requirements satisfied?
**[NOTICE] Bug fixes or features added to Salt require tests.**

- [ ] Docs
- [ ] Changelog - https://docs.saltproject.io/en/master/topics/development/changelog.html
- [x] Tests written/updated

### Commits signed with GPG?
Yes

### CODE REVIEW

File: tests/pytests/scenarios/performance/test_performance.py [hunk 1]
```diff
@@ -0,0 +1,416 @@
...
+        "{}.sls".format(sls_name), sls_contents.format(path=str(location)), state_tree
+    ):
+        yield sls_name
+
+
+def _wait_for_stdout(expected, func, *args, timeout=120, **kwargs):
+    start = time.time()
+    while time.time() < start + timeout:
+        ret = func(*args, **kwargs)
+        print(ret)
```
Comment (Ch3LL) [inline tests/pytests/scenarios/performance/test_performance.py:327]: can this print be removed?

**Review by Ch3LL — CHANGES_REQUESTED** [review]

File: tests/pytests/scenarios/performance/test_performance.py (hunk 1 above)
Comment (MKLeb) [inline tests/pytests/scenarios/performance/test_performance.py:327]:
```suggestion
```

**Review by Ch3LL — DISMISSED** [review]

**Comment by MKLeb** [comment]
re-run all

**Comment by MKLeb** [comment]
re-run all

**Comment by MKLeb** [comment]
re-run all

**Comment by MKLeb** [comment]
re-run all

**Review by Ch3LL — DISMISSED** [review]

**Comment by MKLeb** [comment]
re-run full all

**Comment by MKLeb** [comment]
re-run pull all

**Comment by MKLeb** [comment]
re-run full all

**Review by Ch3LL — APPROVED** [review]

**Pull request merged**